src/extractor/guidance/turn_analysis.cpp

```diff
@@ -95,6 +98,111 @@ inline bool isRampClass(EdgeID eid, const util::NodeBasedDynamicGraph &node_base
+                {
+                    return SPECIAL_EDGEID;
+                }
+
+                roundabout_name_id = edge_data.name_id;
+
+                continue_edge = edge;
+            }
+            else if (!edge_data.roundabout)
+            {
+                // remember all connected road names
+                connected_names.insert(edge_data.name_id);
+            }
+        }
+        return continue_edge;
+    };
+
+    // the roundabout radius has to be the same for all locations we look at it from
+    // to guarantee this, we search the full roundabout for its vertices
+    // and select the three smalles ids
+    std::set<NodeID> roundabout_nodes; // needs to be sorted
+
+    // this value is a hard abort to deal with potential self-loops
+    NodeID last_node = nid;
+    while (0 == roundabout_nodes.count(last_node))
+    {
+        roundabout_nodes.insert(last_node);
+        const auto eid = getNextOnRoundabout(last_node);
+
+        if (eid == SPECIAL_EDGEID)
+        {
+            util::SimpleLogger().Write(logDEBUG) << "Non-Loop Roundabout found.";
+            return false;
+        }
+
+        last_node = node_based_graph.GetTarget(eid);
+    }
+
+    // do we have a dedicated name for the rotary, if not its a roundabout
+    // This function can theoretically fail if the roundabout name is partly
+    // used with a reference and without. This will be fixed automatically
+    // when we handle references separately or if the useage is more consistent
+    if (roundabout_name_id == 0 || connected_names.count(roundabout_name_id))
+    {
+        return false;
+    }
+
+    if (roundabout_nodes.size() <= 1)
+    {
+        return false;
+    }
+
+    // calculate the radius of the roundabout/rotary. For two coordinates, we assume a minimal
+    // circle
+    // with both vertices right at the other side (so half their distance in meters).
+    // Otherwise, we construct a circle through the first tree vertices.
+    auto node_itr = roundabout_nodes.begin();
+    const double radius = roundabout_nodes.size() >= 3
+                              ? util::coordinate_calculation::circleRadius(
+                                    getCoordinate(*node_itr++),
+                                    getCoordinate(*node_itr++),
+                                    getCoordinate(*node_itr))
+                              : 0.5 * util::coordinate_calculation::haversineDistance(
+                                          getCoordinate(*node_itr++),
+                                          getCoordinate(*node_itr));
+
+    // check whether the circle computation has gone wrong
+    // The radius computation can result in infinity, if the three coordinates are non-distinct.
+    // To stay on the safe side, we say its not a rotary
+    if (std::isinf(radius))
+        return false;
+
+    const double constexpr MAX_ROUNDABOUT_RADIUS = 15; // 30 m diameter as final distinction
```

Comment:
We hard-code a ton of heuristic constants. Maybe it's time to refactor them into a `guidance_constants.hpp` header (or something similar) so that we have a single location to take a look at?